chart/templates/scheduler/scheduler-deployment.yaml

```
@@ -299,11 +301,12 @@ spec:
         {{- else if .Values.dags.gitSync.enabled }}
         - name: dags
           emptyDir: {{- toYaml (default (dict) .Values.dags.gitSync.emptyDirConfig) | nindent 12 }}
+        {{- end }}
         {{- if .Values.dags.gitSync.sshKeySecret }}
```

```suggestion
        {{- if and $localOrDagProcessorDisabled .Values.dags.gitSync.enabled .Values.dags.gitSync.sshKeySecret }}
```

It think this, in addition to your other changes in this file, is the right fix.